lib/linter/linter.js

```diff
@@ -779,6 +829,22 @@ function runRules(sourceCode, configuredRules, ruleMapper, parserOptions, parser
     return lintingProblems;
 }
 
+/**
+ * Ensure the source code to be a string.
+ * @param {string|SourceCode} textOrSourceCode The text or source code object.
+ * @returns {string} The source code text.
+ */
+function ensureText(textOrSourceCode) {
+    if (typeof textOrSourceCode === "object") {
+        const { hasBOM, text } = textOrSourceCode;
+        const bom = hasBOM ? "\uFEFF" : "";
+
+        return bom + text;
```

Comment:
Do we want to do this? This will create an 1 character offset in the error message, and `postprocessor` will not be able to combine error messages back to the right location.

Reply:
Yes, because it's the current behavior. The first parameter of `Linter#verify()` method is a string or a `SourceCode` object. If it's a string, the string contains BOM.

But I agree that the BOM is not useful. We can change the behavior on the ESLint 7.0.0 line.